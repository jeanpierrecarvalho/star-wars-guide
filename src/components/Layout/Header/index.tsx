import React from 'react'
import Image from 'next/image'
import Link from 'next/link'

const Header: React.FC = () => {
  return (
    <header
      className="flex flex-row justify-center"
      role="banner"
    >
      <Link
        href="/"
        passHref
        aria-label="Go to homepage"
      >
        <Image
          src="/images/logo.png"
          width={200}
          height={200}
          alt="Star Wars Logo"
          priority={true}
        />
      </Link>
    </header>
  )
}

export default Header
