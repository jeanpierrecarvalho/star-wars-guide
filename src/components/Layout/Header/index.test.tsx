import { render } from '@testing-library/react';
import { vi, describe, it, expect } from 'vitest';

import Header from '@/components/Layout/Header';

vi.mock('next/image', () => ({
	_esModule: true,
	default: ({ src, alt }: { src: string; alt: string }) => {
		return <img src={src} alt={alt} />;
	}
}));

describe('Header Component', () => {
	it('matches the snapshot', () => {
		const { container } = render(<Header />);
		expect(container).toMatchSnapshot();
	});
});
