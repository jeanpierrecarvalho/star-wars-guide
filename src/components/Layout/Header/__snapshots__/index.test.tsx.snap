// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header Component > matches the snapshot 1`] = `
<div>
  <header
    class="flex flex-row justify-center"
    role="banner"
  >
    <a
      aria-label="Go to homepage"
      href="/"
    >
      <img
        alt="Star Wars Logo"
        src="/images/logo.png"
      />
    </a>
  </header>
</div>
`;
